notebooks/feature_engineering/raw/ex1.ipynb

```diff
@@ -0,0 +1,412 @@
+    "\n",
+    "In this exercise, you will develop a baseline model for predicting if a customer will buy the app after clicking through from an ad. With this baseline model, you'll be able to see how your feature engineering and selection efforts improve the model's performance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up code checking\n",
+    "from learntools.core import binder\n",
+    "binder.bind(globals())\n",
+    "from learntools.feature_engineering.ex1 import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "click_data = pd.read_csv('../input/feature-engineering-data/train_sample.csv', \n",
+    "                         parse_dates=['click_time'])\n",
+    "click_data.head(10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Baseline Model\n",
+    "\n",
+    "The first thing you need to do is construct a baseline model. All new features, processing, encodings, and feature selection should improve upon this baseline model. First you need to do a bit of feature engineering before training the model itself.\n",
+    "\n",
+    "### 1) Exercise: Features from timestamps\n",
+    "From the timestamps, create features for the day, hour, minute and second. Store these as new integer columns `day`, `hour`, `minute`, and `second` and a new DataFrame `clicks`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add new columns for timestamp features day, hour, minute, and second\n",
+    "clicks = ____"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%%RM_IF(PROD)%%\n",
+    "# My solution, there are a lot of ways to do this\n",
+    "\n",
+    "# Split up the times\n",
+    "click_times = click_data['click_time']\n",
+    "clicks = click_data.assign(day=click_times.dt.day.astype('uint8'),\n",
+    "                           hour=click_times.dt.hour.astype('uint8'),\n",
+    "                           minute=click_times.dt.minute.astype('uint8'),\n",
+    "                           second=click_times.dt.second.astype('uint8'))\n",
+    "q_1.check()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Uncomment these if you need guidance\n",
+    "# q_1.hint()\n",
+    "# q_1.solution()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run this to check your work\n",
+    "q_1.check()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2) Exercise: Label Encoding\n",
```

Comment:
For users who don't remember the name LabelEncoder, it'd be useful to link back to the explanation in Intermediate ML

Reply:
They'll see this in the first tutorial